crates/ruff_linter/src/rules/pycodestyle/rules/literal_comparisons.rs

```diff
@@ -102,39 +102,47 @@ impl AlwaysFixableViolation for NoneComparison {
 ///
 /// [PEP 8]: https://peps.python.org/pep-0008/#programming-recommendations
 #[violation]
-pub struct TrueFalseComparison(bool, EqCmpOp);
+pub struct TrueFalseComparison {
+    value: bool,
+    op: EqCmpOp,
+    cond: Option<String>,
```

Comment:
I think we should use `SourceCodeSnippet` for this -- can you give that a try? It will ensure that we don't use the name if it's too long or spans multiple lines, since that tends to yield poor diagnostic messages.

Reply:
Sure, updated the branch.

Reply:
Thanks!